src/reflect/scala/reflect/internal/transform/Erasure.scala

```diff
@@ -151,7 +151,10 @@ trait Erasure {
         else if (sym.isDerivedValueClass) eraseDerivedValueClassRef(tref)
         else if (isDottyEnumSingleton(sym)) apply(intersectionType(tp.parents)) // TODO [tasty]: dotty enum singletons are not modules.
         else if (sym.isClass) eraseNormalClassRef(tref)
-        else apply(sym.info.asSeenFrom(pre, sym.owner)) // alias type or abstract type
+        else sym.attachments.get[OpaqueTypeAlias] match {
```

Comment:
Here we are using a symbol attachment to check is the symbol is an opaque type and extract its alias